File: domain/src/main/java/com/arturogutierrez/openticator/domain/otp/time/RemainingTimeCalculator.java
package com.arturogutierrez.openticator.domain.otp.time;

import javax.inject.Inject;

public class RemainingTimeCalculator {

  private final CurrentTimeProvider timeProvider;

  @Inject
  public RemainingTimeCalculator(CurrentTimeProvider timeProvider) {
    this.timeProvider = timeProvider;
  }

  public int calculateRemainingSeconds(long validUntilInSeconds) {
    int remainingSeconds = (int) (validUntilInSeconds - timeProvider.getCurrentTimeInSeconds());
    if (remainingSeconds < 0) {
      remainingSeconds = 0;
    }

    return remainingSeconds;
  }
}

File: domain/src/main/kotlin/com/arturogutierrez/openticator/domain/otp/time/RemainingTimeCalculator.kt
package com.arturogutierrez.openticator.domain.otp.time

import javax.inject.Inject

class RemainingTimeCalculator @Inject constructor(private val timeProvider: TimeProvider) {

  fun calculateRemainingSeconds(validUntilInSeconds: Long): Long {
    val remainingSeconds = (validUntilInSeconds - timeProvider.currentTimeInSeconds()).toInt()
    if (remainingSeconds < 0) {
      return 0
    }

    return remainingSeconds.toLong()
  }
}
